handler.go

```diff
@@ -4,22 +4,38 @@ import (
 	"encoding/json"
 	"log"
 	"net/http"
+	"strconv"
 )
 
 func courseSimpleSearchHandler(w http.ResponseWriter, r *http.Request) {
-	q := r.URL.Query()
-	courseName := q.Get("course_name")
-	courseNameFilterType := q.Get("course_name_filter_type")
-	courseOverview := q.Get("course_overview")
-	courseOverviewFilterType := q.Get("course_overview_filter_type")
-	filterType := q.Get("filter_type")
-	limit := q.Get("limit")
-	offset := q.Get("offset")
 
-	// if !(filterType == "and" || filterType == "or") {
-	// 	w.WriteHeader(http.StatusBadRequest)
-	// 	return
-	// }
+	//Validate request
+	if r.Method != "POST" {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if r.Header.Get("Content-Type") != "application/json" {
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
```suggestion
		w.WriteHeader(http.StatusUnsupportedMediaType)
```


「415 Unsupported Media Type」にあたりそう
- [415 Unsupported Media Type - HTTP | MDN](https://developer.mozilla.org/ja/docs/Web/HTTP/Status/415)
- [rest - HTTP status code for unaccepted Content-Type in request - Stack Overflow](https://stackoverflow.com/questions/11973813/http-status-code-for-unaccepted-content-type-in-request)